### DISCUSSION TITLE
JSON current conditions and windgusts

### DISCUSSION THREAD

**Discussion by cybermaus** [chubin/wttr.in#738] [Ideas]
Would it be possible to add Windgust to the current conditions in the JSON reply we get from [this example link](https://wttr.in/51.691944,4.437778?format=j1).

It is in the hourly list, but that is not trivial to retrieve, but it is not in the current conditions list.

Many thanks
 ```
{
    "current_condition": [
        {
...
            "winddir16Point": "SW",
            "winddirDegree": "230",
            "windspeedKmph": "20",
            "windspeedMiles": "13"
        }
    "weather": [
...
            "hourly": [
                {
...
                    "WindGustKmph": "26",
                    "WindGustMiles": "16",
...
                    "winddir16Point": "SW",
                    "winddirDegree": "227",
                    "windspeedKmph": "13",
                    "windspeedMiles": "8"
...
```

**Comment by chubin** [comment]
I think it is a good idea. Maybe we should add not only windgusts to the current condition, but merge everything we have in the forecast section (for the current time), or add it as a nested dict? Something like `details` or `forecast` ?

**Comment by cybermaus** [comment]
Well to be honest, the idea wasto be able to get current conditions from an underpowered [device like Tasmota](https://github.com/arendst/Tasmota/discussions/15705). And that has very limited JSON pasing abilities.

In fact, by now I learned Tasmote [cannot do arrays of objects](https://github.com/arendst/Tasmota/discussions/15655) (only arrays of values) so indeed it would be needed to create a non-array variant of 'current_conditions'

For now, I got it to work with a different provider then wttr

**Comment by chubin** [comment]
It is a good news that your problem is solved, but it would be really great from my perspective if we could manage to work it with wttr.in too.
So probably, the ideal solution in your case would like a simplified JSON output, without nested dicts, that could pe requested with a special query? Something like `format=j2` ?

**Comment by cybermaus** [comment]
For Tasmota specifically, nested is OK, as long as it is not nested arrays, or any non-atomic structures inside arrays (I may not be using the correct JSON vocabulary)

But generalising it, yes, for simple IoT devices, and simple programming within those devices, a `format=j2` and/or `q=current_conditions` would be ideal.
